dvc/scm/git.py

```diff
@@ -491,3 +497,160 @@ def resolve_commit(self, rev):
         if isinstance(commit, TagObject):
             commit = commit.object
         return commit
+
+    def _backend_func(self, name, *args, **kwargs):
+        for backend in self.backends:
+            try:
+                func = getattr(backend, name)
+                return func(*args, **kwargs)
+            except (AttributeError, NotImplementedError):
+                pass
+        raise NoGitBackendError(name)
+
+    _ignored = partialmethod(_backend_func, "_ignored")
+    set_ref = partialmethod(_backend_func, "set_ref")
+    get_ref = partialmethod(_backend_func, "get_ref")
+    remove_ref = partialmethod(_backend_func, "remove_ref")
+    push_refspec = partialmethod(_backend_func, "push_refspec")
+    fetch_refspecs = partialmethod(_backend_func, "fetch_refspecs")
+    _stash_iter = partialmethod(_backend_func, "_stash_iter")
+    _stash_push = partialmethod(_backend_func, "_stash_push")
+    _stash_apply = partialmethod(_backend_func, "_stash_apply")
+    reflog_delete = partialmethod(_backend_func, "reflog_delete")
```

Comment:
I originally had this set up using class introspection and `setattr` calls in `__init__()`, but when using runtime instrospection, mypy and pylint will end up complaining about missing attributes any time we try to call a backend function. I'm not sure whether or not there's a proper way to get introspection + type checking to work nicely?